lib-injection/auto_inject.rb

```diff
@@ -0,0 +1,84 @@
+return if ENV['skip_autoinject']
```

Comment:
Usually environment variables are set in `ALL_CAPS`, so I'm curious why deviate from this standard?

Also, since this is a ddtrace-wide configuration, perhaps use `DD_TRACE_` the prefix as well?

Finally, consider actually testing the value. Doing only `if ENV['skip_autoinject']` can be confusing because if I set it to `skip_autoinject=no` or `=false`, the auto-inject is still skipped.

All together, my suggestion is to instead use:

```ruby
return if ENV['DD_TRACE_SKIP_AUTOINJECT'] == 'true'
```

(or `['true', 'y', 'yes', '1'].include?(...)` if you'd like to support a few other variants)